lib/Service/SMimeService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use OCA\Mail\Exception\OpensslUnavailableException;
+use OCP\ICertificateManager;
+use OCP\ITempManager;
+
+class SMimeService {
+
+	private ITempManager $tempManager;
+	private ICertificateManager $certificateManager;
+
+	public function __construct(ITempManager $tempManager,
+								ICertificateManager $certificateManager) {
+		$this->tempManager = $tempManager;
+		$this->certificateManager = $certificateManager;
+	}
+
+	/**
+	 * Check whether the openssl module is available.
+	 * It provides the underlying S/MIME apis.
+	 *
+	 * @return bool
+	 */
+	public function hasOpenssl(): bool {
+		return extension_loaded('openssl');
```

Comment:
FYI: php-openssl is a hard requirement for Nextcloud (https://docs.nextcloud.com/server/latest/admin_manual/installation/source_installation.html). Nextcloud don't work without php-openssl :boom:.

Reply:
Thanks for letting me know. I'll get rid of the check.